server/src/main/java/org/opensearch/wlm/ResourceType.java

```diff
@@ -24,19 +26,35 @@
  */
 @PublicApi(since = "2.17.0")
 public enum ResourceType {
-    CPU("cpu", task -> task.getTotalResourceUtilization(ResourceStats.CPU), true),
-    MEMORY("memory", task -> task.getTotalResourceUtilization(ResourceStats.MEMORY), true);
+    CPU("cpu", true, CpuUsageCalculator.INSTANCE, new QueryGroupUsageHelper() {
+        @Override
+        protected double getNormalisedThreshold(QueryGroup queryGroup) {
+            return queryGroup.getResourceLimits().get(ResourceType.CPU) * getSettings().getNodeLevelCpuCancellationThreshold();
+        }
```

Comment:
Wondering if we can have common `getNormalizedThreshold` method within `QueryGroupUsageHelper`, since we should have resource type already. Although, we need to figure out good way for node level resource threshold.

Reply:
I have moved this into the resourceType and rest of this class logic is in TaskCancellar to maintain the coherency of the logic